MiniApp/Classes/RealMiniApp.swift

```diff
@@ -48,6 +48,10 @@ internal class RealMiniApp {
                 }
                 self.downloadMiniApp(appInfo: appInfo, completionHandler: completionHandler)
             case .failure(let error):
+                let createError = error as NSError
+                if createError.code == NSURLErrorNotConnectedToInternet {
+                    self.downloadMiniApp(appInfo: appInfo, completionHandler: completionHandler)
```

Comment:
What case is this for? In case the device was connected to the internet when you fetched the manifest and then disconnected when you try to download the Mini App? But then the `completionHandler` is being immediately called with failure below, so the result here will never get returned to the user, right? Maybe you meant to add a `return` here?

Is there a way to cover this case within `MiniAppDownloader#download` and have it return the cached Mini App? It seems like functionality which should be part of that class and should be covered by the same functionality as when fetching the manifest fails.

Reply:
This method will fetch the mini-app information from the platform and compare if it is the latest version and if not we download the latest one. In the `.failure` block (esp for Offline scenario) I was calling download method because the download manager will take care of checking if there is any cached version available.
Now that I have changed a few things in `MiniAppDownloader`, instead of downloading, I am calling the `getCachedMiniApp` method now. Please check it